netket/stats/mpi_stats.py

```diff
@@ -85,7 +85,8 @@ def sum(a, axis=None, keepdims: bool = False):
         # assume it's a scalar
         a_sum = jnp.asarray(a)
 
-    return mpi_sum(a_sum)
+    out, _ = mpi.mpi_sum_jax(a_sum)
```

Comment:
just for my understanding: why these are called ```mpi_sum_jax``` and not ```mpi_sum```?

Reply:
Because `mpi_sum` is the standard `mpi4py` `mpi_sum` and acts on standard numpy arrays while `mpi_sum_jax` is the `mpi4jax` version.

I started being explicit about it on purpose after i found the deadlock bug #655 in order to avoid such bugs in the future.